src/components/VetList.tsx

```diff
@@ -0,0 +1,81 @@
+import { useEffect, useState } from 'react';
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Paper,
+  TablePagination,
+} from '@material-ui/core';
+import PetService from '../service/PetClinicService';
+
+const VetList = () => {
+  const [vetList, setVetList] = useState([
+    { id: '', firstName: '', lastName: '', specialties: [] },
+  ]);
+
+  const petService = new PetService();
```

Comment:
You should not construct class instances in React components here, because this instances are re-created at every (!) render cycle.

Reply:
thanks Nils! I have pushed an update with exported functions as you mentioned in your other comment.